src/quo/components/buttons/logout_button/view.cljs

```diff
@@ -0,0 +1,32 @@
+(ns quo.components.buttons.logout-button.view
+  (:require
+    [quo.components.buttons.logout-button.style :as style]
+    [quo.components.icon :as icon]
+    [quo.components.markdown.text :as text]
+    [quo.foundations.colors :as colors]
+    [quo.theme :as quo.theme]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [utils.i18n :as i18n]))
+
+(defn- view-internal
+  [_ _]
+  (let [pressed? (reagent/atom false)]
+    (fn
+      [{:keys [on-press on-long-press disabled? theme container-style]}]
+      [rn/pressable
+       {:accessibility-label :log-out-button
+        :on-press            on-press
+        :on-press-in         #(reset! pressed? true)
```

Comment:
@ilmotta if i understand your point correctly and you mean something like this, i can't find it beneficial 

```
(defn- set-pressed [pressed? value]
(reset! pressed? value))

(defn- view-internal
  [_]
  (let [pressed? (reagent/atom false)]
    (fn
      [{:keys [on-press on-long-press disabled? theme container-style]}]
      [rn/pressable
       {:accessibility-label :log-out-button
        :on-press            on-press
        :on-press-in         #(set-pressed pressed? true)
        :on-press-out        #(set-pressed pressed? nil)
```

Reply:
@mohsen-ghafouri, the code you shared still doesn't solve the problem that anonymous functions are being recreated on every render. But as I said, the lack of stable handlers passed to `rn/pressable` is not really a perf problem. I put my comment as a *suggestion* because I think it's definitely up to you to decide if you want to do this here.

Usually when somebody says *move outside the renderer* it means moving outside the inner renderer (as Reagent itself refers to it sometimes https://github.com/reagent-project/reagent/blob/a14faba55e373000f8f93edfcfce0d1222f7e71a/doc/FAQ/ComponentNotRerendering.md#props-change). So we have the component renderer and the inner renderer (the anonymous function), kind of confusing 🤷🏼 

If you allow me, I highly suggest reading this doc page from re-frame about *stable handlers* (yes, it's confusing, it's not in the Reagent docs) https://github.com/day8/re-frame/blob/39adca93673f334dc751ee2d99d340b51a9cc6db/docs/on-stable-dom-handlers.md

But the code would be:

```clojure
(defn- view-internal
  [_]
  (let [pressed?     (reagent/atom false)
        on-press-in  #(reset! pressed? true)
        on-press-out #(reset! pressed? nil)]
    (fn
      [{:keys [on-press on-long-press disabled? theme container-style]}]
      [rn/pressable
       {:accessibility-label :log-out-button
        :on-press            on-press
        :on-press-in         on-press-in
        :on-press-out        on-press-out
        :on-long-press       on-long-press
        :disabled            disabled?
        :style               (merge (style/main {:pressed?  @pressed?
                                                 :theme     theme
                                                 :disabled? disabled?})
                                    container-style)}
       [icon/icon :i/log-out {:color (if pressed? colors/white-opa-40 colors/white-opa-70)}]
       [text/text {:weight :medium :size :paragraph-1}
        (i18n/label :t/logout)]])))
```

The re-frame doc page says:

> This technique is only useful when you are writing performance-critical
components.

Just a reminder that the docs were written with webapps in mind. In our case, helping Reagent skip processing components when props change even in components re-rendering at a low-frequency can be beneficial because the device is already quite busy with all the animations going on, signals being processed, etc.

Reply:
ah now i got it, those links were informative, thank you @ilmotta